### PR TITLE
Added streq and strneq macro

### PR DESCRIPTION
**Pull request by manuelmessner**
Targets #478

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Will you continue here and replace code which can be replaced by this?

**Comment by neithernut** [comment]
I'm unhappy with the choice of name for `srtneq` because it looks like "string _not_ equal".

**Comment by matthiasbeyer** [comment]
We discussed this... I guess it's more consistent with `strncmp` and the like... but I understand your point. Suggestions?

**Comment by manuelmessner** [comment]
@matthiasbeyer 

Yes I'm doing this right now

@neithernut 

I'll replace only the streq() as long we didn't got a name for strneq

**Comment by neithernut** [comment]
`streqn`? Mh... sounds weird.

**Comment by matthiasbeyer** [comment]
Btw I would prefer having it with a `ws_` prefix... we already had a problem with a missing prefix, no need to fall into this trap again...

**Comment by manuelmessner** [comment]
@matthiasbeyer 
Changed. I push it later.

@neithernut
what about this?
nstreq => not_string_equal
strneq => string_equal_num

**Comment by neithernut** [comment]
Mh... `strneq` might be the only feasible option...

**Comment by matthiasbeyer** [comment]
Seems so.

@manuelmessner I wouldn't include `nstreq`, as there is no advantage over doing `!streq(...)` ...

**Comment by manuelmessner** [comment]
Ok can I implement that?
`ws_streq` and `ws_strneq`?

**Comment by matthiasbeyer** [comment]
ACK for the first, ACK for the second if @neithernut ACKs it.

**Comment by neithernut** [comment]
Yep, ACK from me, too.

File: src/util/string.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * @{
+ */
+
+#ifndef __WS_UTIL_STRING_H__
+#define __WS_UTIL_STRING_H__
+
+
+/**
+ * Compare two strings on equality
```
Comment (xVin) [inline src/util/string.h]: "compare sth on" isn't correct English as far as I know. Maybe just write "Check if to strings are equal" if you want to keep it short?

File: src/util/string.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * @return 1 in case of equality, else 0
+ */
+#define streq(x, y) (0 == strcmp((x), (y)))
+
+
+/**
+ * Compare the first n bytes of two strings on equality
```
Comment (xVin) [inline src/util/string.h]: see above.

File: src/util/string.h (hunk 1 above)
Comment (matthiasbeyer) [inline src/util/string.h]: "if **to** strings are equal"? I blame you for this!

Comment (manuelmessner) [inline src/util/string.h]: Haha :D

@xVin Ok I'll change that.

Comment (xVin) [inline src/util/string.h]: lol :D
that's karma

**Comment by manuelmessner** [comment]
ws_strneq is following in a few minutes

**Comment by manuelmessner** [comment]
Pls review

**Comment by neithernut** [comment]
ACK.

**Pull request merged**